numpy/lib/recfunctions.py

```diff
@@ -299,11 +298,9 @@ def _izip_fields(iterable):
     for element in iterable:
         if (hasattr(element, '__iter__') and
                 not isinstance(element, str)):
-            for f in _izip_fields(element):
-                yield f
+            yield from _izip_fields(element)
         elif isinstance(element, np.void) and len(tuple(element)) == 1:
-            for f in _izip_fields(element):
-                yield f
+            yield from _izip_fields(element)
```

Comment:
Note: this is the same statement as the previous block, which is odd. Should this be delegated into a separate issue?

Reply:
Yeah, let's not worry about that in this patch

Reply:
It was like this since the first version of this file. I don't think combining the two `if` clauses with an `or` would make it much cleaner, but I agree it is a bit odd. Maybe a comment noting the strange construct would help.

Reply:
I've added a comment for the second statement.